frontend/app/src/components/home/ChatMessageMenu.svelte

```diff
@@ -260,13 +286,67 @@
             }
         });
     }
+
+    function toggleReaction(reaction: string) {
+        if (canReact) {
+            const kind = client.containsReaction(createdUser.userId, reaction, msg.reactions)
+                ? "remove"
+                : "add";
+
+            client
+                .selectReaction(
+                    chatId,
+                    createdUser.userId,
+                    threadRootMessageIndex,
+                    msg.messageId,
+                    reaction,
+                    createdUser.username,
+                    createdUser.displayName,
+                    kind,
+                )
+                .then((success) => {
+                    if (success && kind === "add") {
+                        client.trackEvent("reacted_to_message");
+
+                        // Note: Manually selected reactions do not increment
+                        // their fav counter by default, so we do it manually.
+                        emojiDb.incrementFavoriteEmojiCount(reaction);
```

Comment:
To make this reactive I think we would _also_ need to re-compute `loadedReactions`. We probably _will_ because as it stands that expression will only be evaluated when this component mounts.